spec/services/update_course_stats_spec.rb

```diff
@@ -78,6 +78,23 @@
       create(:courses_user, course_id: course.id, user_id: user.id)
     end
 
+    it 'tracks individual errors instead of batch count' do
+  
+      allow(Sentry).to receive(:capture_exception)
+      allow_any_instance_of(LiftWingApi).to receive(:get_single_revision_parsed_data).and_raise(Faraday::ConnectionFailed)
+    
+      rev_ids = [1, 2, 3, 4]
+      subject.get_revision_data(rev_ids)  # Assuming `subject` is an instance of LiftWingApi
```

Comment:
subject is defined in line 7; it's in instance of `UpdateCourseStats` initialized with a course.

Reply:
Did you run this test locally? I would expect it to fail with a NoMethodError here, because the subject is not a LiftWingApi instance.

Reply:
> Did you run this test locally? I would expect it to fail with a NoMethodError here, because the subject is not a LiftWingApi instance.

Yes It threw that error, i reverted changes in that `update_course_stats_spec.rb `file , trying to test with the default expected 8 errors but I am getting more than that when i run the test
```
UpdateCourseStats sentry course update error tracking tracks update errors properly in LiftWing
     Failure/Error: expect(course.flags['update_logs'][1]['error_count']).to eq 8
     
       expected: 8
            got: 336
```
Iam thinking there might be an issue with how errors are being counted in the batch processing logic but have not found it yet. Because the error_count is being incremented by a much larger value than expected. In the default test, expected error_count is to be 8, but it turned out to be 336.

Kindly guide me

Reply:
> Did you run this test locally? I would expect it to fail with a NoMethodError here, because the subject is not a LiftWingApi instance.

Yes It threw that error, i reverted changes in that `update_course_stats_spec.rb `file , trying to test with the default expected 8 errors but I am getting more than that when i run the test
```
UpdateCourseStats sentry course update error tracking tracks update errors properly in LiftWing
     Failure/Error: expect(course.flags['update_logs'][1]['error_count']).to eq 8
     
       expected: 8
            got: 336
```
Iam thinking there might be an issue with how errors are being counted in the batch processing logic but have not found it yet. Because the error_count is being incremented by a much larger value than expected. In the default test, expected error_count is to be 8, but it turned out to be 336.

Kindly guide me

Reply:
The other errors are likely from web requests that other parts of the full `UpdateCourseStats` routine, which fail because VCR prevents from them running. Try adding some `puts` statements inside the error logging method to see what is coming through.